src/streams.c

```diff
@@ -769,104 +769,49 @@ Obj FuncPrint (
     return 0;
 }
 
-static Obj PRINT_OR_APPEND_TO(Obj args, int append)
+static Obj PRINT_OR_APPEND_TO_FILE_OR_STREAM(Obj args, int append, int file)
 {
     const char * volatile funcname = append ? "AppendTo" : "PrintTo";
     volatile Obj        arg;
-    volatile Obj        filename;
+    volatile Obj        destination;
     volatile UInt       i;
     syJmp_buf           readJmpError;
 
-    /* first entry is the filename                                         */
-    filename = ELM_LIST(args,1);
-    while ( ! IsStringConv(filename) ) {
-        filename = ErrorReturnObj(
-            "%s: <filename> must be a string (not a %s)",
-            (Int)funcname, (Int)TNAM_OBJ(filename),
-            "you can replace <filename> via 'return <filename>;'" );
+    /* first entry is the file or stream                                   */
+    destination = ELM_LIST(args,1);
+
+    /* try to open the output                                              */
+    if ( file ) {
+        /* the calling functions check that destination is a GAP string,
+        ** but it may not yet be in string representation                  */
+        ConvString(destination);
```

Comment:
This is better than the code before, but it'll still crash if the first argument is not a string. What is the motivation for removing the error checking? Personally, I always prefer to have kernel functions that can't be used to crash GAP (and if we ever want to even remotely consider security and safety, the first thing we'll have to do is to add argument validation to all kernel functions anyway)

Reply:
I've updated this to use `IsStringConv` again. Instead of removing the check, the commit msg now reads:
```
- Changes the error handling if the destination filename is not a
  string. In that case, the function prints an error message and
  quits. Previously the user could give another filename.
```